### PR TITLE
FR-13417 - Fixed include query param after signup

### PR DESCRIPTION
**Pull request by yuvalotem1**
## The Reported Issue
Dev success reported that `includeQueryParam: true` is not working after signup and used to work in older versions.

## The Real Issue
After digging deeper, I discovered that this bug is unrelated to `includeQueryParam`.
The reason we keep the query param after sign-up is because of `queryKeeper` hook in the react SDK.

## The Cause of the bug
The reason for the bug was that we removed the logic of query keeper into the provider. Therefore it won't be called on the login page because the provider hides the children if the user is not authenticated.

## The Fix
1. Remove `QueryKeeperWrapper` to be under `alwaysVisibleChildren` to make it work as before 66dfe7f343d353777326764e6fcb7c3eab44bfce
2. Refactor alwaysVisibleChildren to different component c8723c3851ce9f097b4594609f33ce0284d5de53

## Demo

https://github.com/frontegg/frontegg-react/assets/106734943/80be80a7-bd4d-4f8d-9c85-9ed7ab9cade6

### CODE REVIEW

File: packages/react/rollup.config.js [hunk 1]
```diff
@@ -51,6 +51,7 @@ const cjsPlugins = [
 const isExternal = (id) => {
   return (
     id !== './FronteggProvider' &&
+    id !== './AlwaysRenderInProvider' &&
```
Comment (doregg) [inline packages/react/rollup.config.js:54]: why?

**Review by doregg — DISMISSED** [review]
LGTM
Did u check how it affects other pages / scenarios?
I'm just thinking what was the reason to move it to be inside the provider...

The code looks good technically

File: packages/react/rollup.config.js (hunk 1 above)
Comment (yuvalotem1) [reply packages/react/rollup.config.js:54]: We need to add every new file to include it in the bundle :(

File: babel.config.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
```
Comment (doregg) [inline babel.config.js:1]: isn't it affect the build somehow? it looks like a file for babel not only for testing...

File: package.json [hunk 3]
```diff
@@ -7,16 +7,25 @@
     "prettier-hook": "pretty-quick",
     "prettier-check-hook": "pretty-quick --check",
     "prepublishOnly": "make move-package-json-to-dist",
-    "update-version": "make update-entry"
+    "update-version": "make update-entry",
+    "test": "jest --env=jsdom"
```
Comment (doregg) [inline package.json:11]: why do u need jsdom? what does it mean?

File: package.json [hunk 4]
```diff
@@ -7,16 +7,25 @@
...
   },
   "devDependencies": {
+    "@babel/preset-env": "^7.23.3",
```
Comment (doregg) [inline package.json:14]: what I don't like is that we can't run tests.. it's a big change adding babel and all libs
Does babel addition affect the build somehow?

File: package.json [hunk 5]
```diff
@@ -44,5 +53,10 @@
   },
   "workspaces": [
     "packages/*"
-  ]
+  ],
+  "jest": {
+    "transformIgnorePatterns": [
+      "node_modules/(?!@ngrx|(?!deck.gl)|ng-dynamic)"
```
Comment (doregg) [inline package.json:59]: ngrx? is for angular also ng-fynamic.. what is deck.gl? do we need it?

File: packages/react/tsconfig.json [hunk 6]
```diff
@@ -2,11 +2,13 @@
   "extends": "../../tsconfig.json",
   "compilerOptions": {
     "isolatedModules": false,
-    "outDir": "./dist"
+    "outDir": "./dist",
+    "types": ["jest"]
```
Comment (doregg) [inline packages/react/tsconfig.json:6]: what does it mean?

File: packages/react/tsconfig.json [hunk 7]
```diff
@@ -2,11 +2,13 @@
...
+
   },
   "include": [
     "./src/**/*.ts"
-  ],
+, "src/QueeryKeeper.spec.tsx"  ],
```
Comment (doregg) [inline packages/react/tsconfig.json:11]: what does it mean?

File: package.json [hunk 8]
```diff
@@ -7,16 +7,25 @@
...
-    "update-version": "make update-entry"
+    "update-version": "make update-entry",
+    "test": "jest --env=jsdom"
   },
   "devDependencies": {
+    "@babel/preset-env": "^7.23.3",
+    "@babel/preset-react": "^7.23.3",
+    "@babel/preset-typescript": "^7.23.3",
     "@rollup/plugin-node-resolve": "^8.0.1",
+    "@testing-library/react": "12.1.2",
```
Comment (doregg) [inline package.json:18]: do we need all libs in dep? why not devdep?

File: package.json (hunk 3 above)
Comment (yuvalotem1) [reply package.json:11]: we need to access the dom during the test because we want to render a component

File: babel.config.js (hunk 2 above)
Comment (yuvalotem1) [reply babel.config.js:1]: I think create react app uses preset babel config, this config is used to run the jest test

Comment (yuvalotem1) [reply babel.config.js:1]: Also our react sdk is using roll for build
image

File: package.json (hunk 5 above)
Comment (yuvalotem1) [reply package.json:59]: not sure, just regex I found on web to allow the esm in jest tests

File: packages/react/tsconfig.json (hunk 6 above)
Comment (yuvalotem1) [reply packages/react/tsconfig.json:6]: To declare the jest functions for ts and avoid errors in the editor like this -
image

File: packages/react/tsconfig.json (hunk 7 above)
Comment (yuvalotem1) [reply packages/react/tsconfig.json:11]: was generated automatically after `yarn test` command

File: package.json (hunk 8 above)
Comment (yuvalotem1) [reply package.json:18]: its on devdep

File: package.json (hunk 4 above)
Comment (yuvalotem1) [reply package.json:14]: this is only on the global level
We have our react SDK inside `react` folder that has its own build tools and rules

**Review by doregg — APPROVED** [review]
It looks like you know what you did, but I would check that the bundle looks the same before and after and create an alpha version to check it.
@rotemzif1 @yuvalotem1 WDYT?

**Pull request merged**